urllib3/util/retry.py

```diff
@@ -123,18 +124,28 @@ class Retry(object):
     :param tuple history: The history of the request encountered during
         each call to :meth:`~Retry.increment`. The list is in the order
         the requests occurred. Each list item is of class :class:`RequestHistory`.
+
+    :param iterable status_retry_after_list:
+        A set of integer HTTP status codes that we should retry on if
+        Retry-After response header exists.
+
+        By default, we only retry on status codes defined as
+        :attr:`Retry.DEFAULT_RETRY_AFTER_STATUS_CODES`.
+
     """
 
     DEFAULT_METHOD_WHITELIST = frozenset([
         'HEAD', 'GET', 'PUT', 'DELETE', 'OPTIONS', 'TRACE'])
 
+    DEFAULT_RETRY_AFTER_STATUS_CODES = frozenset([413, 429, 503])
+
     #: Maximum backoff time.
     BACKOFF_MAX = 120
 
     def __init__(self, total=10, connect=None, read=None, redirect=None,
                  method_whitelist=DEFAULT_METHOD_WHITELIST, status_forcelist=None,
                  backoff_factor=0, raise_on_redirect=True, raise_on_status=True,
-                 history=None):
+                 history=None, status_retry_after_list=DEFAULT_RETRY_AFTER_STATUS_CODES):
```

Comment:
I'd like an option to disable the Retry-After header respecting completely and in a way that makes sense - doing `status_retry_after_list = []` should work, but it's not ideal. Something like `respect_retry_after_headers=True` as a default would be ideal.